src/power_grid_model/core/data_handling.py

```diff
@@ -139,7 +137,7 @@ def create_output_data(
     for name, count in all_component_count.items():
         # shape
         if is_batch:
-            shape: Union[Tuple[int], Tuple[int, int]] = (batch_size, count)
+            shape: Tuple[int] | Tuple[int, int] = (batch_size, count)
```

Comment:
please double-check that all cases of `List`, `Tuple`, `Set`, `Dict` have been removed

```suggestion
            shape: tuple[int] | tuple[int, int] = (batch_size, count)
```